rai/_cli/propagate.py

```diff
@@ -18,41 +19,56 @@
 import pathlib
 import re
 import textwrap
-from typing import Literal, cast
+from typing import Dict, cast
 
 import black
 import tomlkit
+from typing_extensions import Literal
 
-from ._paths import DOCS_DIR, LIBRARY_DIR, REPO_ROOT
+from rai._paths import RAICONTOURS_REPO_ROOT, REPO_ROOT
 
-PyProjectContents = dict[
-    Literal["tool"], dict[Literal["poetry"], dict[Literal["version"], str]]
```

Comment:
These changes from `dict` -> `Dict` were so that I could support as far back as Python 3.7.